### PR TITLE
Roald/move opname and resource around

### PR DESCRIPTION
**Pull request by RmStorm**

### CODE REVIEW

File: troncos/traces/decorate.py [hunk 1]
```diff
@@ -99,6 +103,8 @@ def myfunc1()
     def myfunc2()
         return "This will be traced using a custom provider"
     """
+    if resource:
+        attributes["resource"] = resource
```
Comment (gbbirkisson) [inline troncos/traces/decorate.py:107]: Attributes can be None. Just pass `resource` into `_trace_function` below and be done with it.

File: troncos/traces/decorate.py [hunk 2]
```diff
@@ -126,13 +134,17 @@ def trace_block(
     with trace_block(name="my block", attributes={"some": "attribute"}):
         time.sleep(1)
     """
+    if resource:
+        attributes["resource"] = resource
```
Comment (gbbirkisson) [inline troncos/traces/decorate.py:138]: Attributes can be None

File: troncos/traces/decorate.py [hunk 3]
```diff
@@ -161,6 +173,9 @@ def m3(self):
     """
 
     def _dec(cls: Type[TClass]) -> Type[TClass]:
+        if resource:
+            attributes["resource"] = resource
```
Comment (gbbirkisson) [inline troncos/traces/decorate.py:177]: Attributes can be None. Just pass `resource` into `_trace_function` below and be done with it.

File: troncos/traces/decorate.py [hunk 4]
```diff
@@ -213,6 +229,8 @@ def my_function():
 
     # End of module
     """
+    if resource:
+        attributes["resource"] = resource
```
Comment (gbbirkisson) [inline troncos/traces/decorate.py:233]: Attributes can be None. Just pass `resource` into `_trace_function` below and be done with it.

**Review by gbbirkisson — CHANGES_REQUESTED** [review]

File: troncos/traces/decorate.py [hunk 5]
```diff
@@ -115,7 +121,9 @@ def _inner(f: Callable[P, R]) -> Callable[P, R]:
 
 
 def trace_block(
-    name: str,
+    name: str | None = None,
+    resource: str | None = None,
+    *,
```
[the remark this answers is not in the document]
Comment (RmStorm) [reply troncos/traces/decorate.py:127]: to force attribute to be a named parameter :sweat_smile: Seemed like a good idea

File: troncos/traces/decorate.py [hunk 6]
```diff
@@ -99,40 +107,46 @@ def myfunc1()
...
     attributes: Attributes | None = None,
     tracer_provider: opentelemetry.trace.TracerProvider | None = None,
 ) -> _GeneratorContextManager[opentelemetry.trace.Span]:
     """
     Trace using a with statement. You can supply a tracer provider, if none is supplied,
     the global tracer provider will be used. Example:
 
-    with trace_block(name="my block", attributes={"some": "attribute"}):
+    with trace_block("cool.block", "data!", attributes={"some": "attribute"}):
```
Comment (gbbirkisson) [inline troncos/traces/decorate.py:135]: Now this makes total sense. You have an operation name that is acting on some resource :+1: 

How does this make sense on the `trace_function` decorator? Is `data!`, aka `resource` dynamic? 

Can you add a similar example to the decorator docs?

File: troncos/traces/decorate.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (RmStorm) [reply troncos/traces/decorate.py:127]: It's not possible for the ones that have `*args` but I will add it to the other ones.

File: pyproject.toml [hunk 7]
```diff
@@ -21,6 +21,7 @@ opentelemetry-propagator-jaeger = "^1.12.0"
 opentelemetry-exporter-otlp-proto-http = "^1.13.0"
 backoff = ">=1.0,<2.0"
 # Above: backoff is pinned to <2.0 because https://github.com/open-telemetry/opentelemetry-python/issues/2829
+types-requests = "^2.28.11.2"
```
Comment (gbbirkisson) [inline pyproject.toml:24]: This should be a dev dependency, and it already is: https://github.com/kolonialno/troncos/blob/main/pyproject.toml#L37

File: .flake8 [hunk 8]
```diff
@@ -1,2 +1,3 @@
 [flake8]
-max-line-length = 88
\ No newline at end of file
+max-line-length = 88
+ignore = B009,W503
```
Comment (gbbirkisson) [inline .flake8:3]: What is the reason for these ignores? Is it bothering you in a lot of places?

File: README.md [hunk 9]
```diff
@@ -293,8 +293,8 @@ Trace using a with statement. You can supply a tracer provider, if none is suppl
 ```python
 from troncos.traces.decorate import trace_block
 
-with trace_block(name="my block", attributes={"some": "attribute"}):
-    print("... do something ...")
+with trace_block(name="my.action", resource="some thing", attributes={"some": "attribute"}):
+    print("... do an action to a  thing...")
```
Comment (gbbirkisson) [inline README.md:297]:
```suggestion
    print("... do an action to a thing...")
```

**Review by gbbirkisson — APPROVED** [review]

**Pull request merged**